packages/calcite-components/src/components/tab-nav/tab-nav.scss

```diff
@@ -1,78 +1,199 @@
 :host {
+  --calcite-internal-tab-nav-gradient-start-side: left;
+  --calcite-internal-tab-nav-gradient-end-side: right;
+
   @apply relative flex;
 }
 
 .scale-s {
+  --calcite-internal-tab-nav-scroller-button-width: #{$calcite-size-24};
   min-block-size: theme("spacing.6");
 }
 
 .scale-m {
+  --calcite-internal-tab-nav-scroller-button-width: #{$calcite-size-32};
   min-block-size: theme("spacing.8");
 }
 
 .scale-l {
+  --calcite-internal-tab-nav-scroller-button-width: #{$calcite-size-44};
   min-block-size: theme("spacing.11");
 }
 
-:host([layout="center"]:not([bordered])) {
-  // `tab-nav` in all scales in layout="center" has a padding of 20px on both ends
-  padding-inline: theme("margin.5");
+.calcite--rtl {
+  --calcite-internal-tab-nav-gradient-start-side: right;
+  --calcite-internal-tab-nav-gradient-end-side: left;
+}
 
-  //override margin-inline-end for the last child for tab-nav to implement the 20px padding on both ends instead
-  .tab-nav {
-    ::slotted(calcite-tab-title:last-child) {
-      margin-inline-end: theme("margin.0");
-    }
+$last-mask-color-stop-position: 51%; // we go beyond the half point to ensure the mask color stops overlap when both start and end are overflowing
+
+.tab-nav--start-overflow {
+  .tab-nav-active-indicator-container,
+  .tab-titles-slot-wrapper {
+    mask-image: linear-gradient(
+      to var(--calcite-internal-tab-nav-gradient-end-side),
+      transparent,
+      transparent var(--calcite-internal-tab-nav-scroller-button-width),
+      white var(--calcite-internal-tab-nav-scroller-button-width),
+      white $last-mask-color-stop-position
+    );
+  }
+}
+
+.tab-nav--end-overflow {
+  .tab-nav-active-indicator-container,
+  .tab-titles-slot-wrapper {
+    mask-image: linear-gradient(
+      to var(--calcite-internal-tab-nav-gradient-start-side),
+      transparent,
+      transparent var(--calcite-internal-tab-nav-scroller-button-width),
+      white var(--calcite-internal-tab-nav-scroller-button-width),
+      white $last-mask-color-stop-position
+    );
+  }
+}
+
+.tab-nav--start-overflow.tab-nav--end-overflow {
+  .tab-nav-active-indicator-container,
+  .tab-titles-slot-wrapper {
+    mask-image: linear-gradient(
+        to var(--calcite-internal-tab-nav-gradient-end-side),
+        transparent,
+        transparent var(--calcite-internal-tab-nav-scroller-button-width),
+        white var(--calcite-internal-tab-nav-scroller-button-width),
+        white $last-mask-color-stop-position,
+        transparent $last-mask-color-stop-position
+      ),
+      linear-gradient(
+        to var(--calcite-internal-tab-nav-gradient-start-side),
+        transparent,
+        transparent var(--calcite-internal-tab-nav-scroller-button-width),
+        white var(--calcite-internal-tab-nav-scroller-button-width),
+        white $last-mask-color-stop-position,
+        transparent $last-mask-color-stop-position
+      );
+  }
+}
+
+.tab-nav::-webkit-scrollbar {
```

Comment:
Whats this for? we're hiding the scrollbar but still allowing scrolling?

Reply:
It was added [here](https://github.com/Esri/calcite-design-system/commit/712fb31655e851973502737764588881b73b4e3c). Removing it doesn't seem to have any effect. @Elijbet Do you recall why this is needed? If not, I can drop this. Good catch! ✨🎣✨